pex/scie/__init__.py

```diff
@@ -0,0 +1,175 @@
+        ),
+    )
+    parser.add_argument(
+        "--scie-platform",
+        dest="scie_platforms",
+        default=[],
+        action="append",
+        type=SciePlatform.for_value,
+        choices=SciePlatform.values(),
+        help=(
+            "The platform to produce the native PEX scie executable for. Can be specified multiple "
+            "times."
+        ),
+    )
+    parser.add_argument(
+        "--scie-pbs-release",
+        dest="scie_pbs_release",
+        default=None,
+        type=str,
+        help=(
+            "The Python Standalone Builds release to use. Currently releases are dates of the form "
+            "YYYYMMDD, e.g.: '20240713'. See their GitHub releases page at "
+            "https://github.com/indygreg/python-build-standalone/releases to discover available "
+            "releases. If left unspecified the latest release is used. N.B.: The latest lookup is "
+            "cached for 5 days. To force a fresh lookup you can remove the cache at "
+            "<USER CACHE DIR>/science/downloads."
+        ),
+    )
+    parser.add_argument(
+        "--scie-python-version",
+        dest="scie_python_version",
+        default=None,
+        type=Version,
+        help=(
+            "The portable CPython version to select. Can be either in `<major>.<minor>` form; "
+            "e.g.: '3.11', or else fully specified as `<major>.<minor>.<patch>`; e.g.: '3.11.3'. "
+            "If you don't specify this option, Pex will do its best to guess appropriate portable "
+            "CPython versions. N.B.: Python Standalone Builds does not provide all patch versions; "
+            "so you should check their releases at "
+            "https://github.com/indygreg/python-build-standalone/releases if you wish to pin down "
+            "to the patch level."
+        ),
+    )
+
+
+def render_options(options):
+    # type: (ScieOptions) -> str
+
+    args = ["--scie", str(options.style)]
+    for platform in options.platforms:
+        args.append("--scie-platform")
+        args.append(str(platform))
+    if options.pbs_release:
+        args.append("--scie-pbs-release")
+        args.append(options.pbs_release)
+    if options.python_version:
+        args.append("--scie-python-version")
+        args.append(".".join(map(str, options.python_version)))
+    return " ".join(args)
+
+
+def extract_options(options):
+    # type: (Namespace) -> Optional[ScieOptions]
+
+    if not options.scie_style:
+        return None
+
+    python_version = None  # type: Optional[Union[Tuple[int, int], Tuple[int, int, int]]]
+    if options.scie_python_version:
+        if (
+            not options.scie_python_version.parsed_version.release
+            or len(options.scie_python_version.parsed_version.release) < 2
+        ):
+            raise ValueError(
+                "Invalid Python version: '{python_version}'.\n"
+                "Must be in the form `<major>.<minor>` or `<major>.<minor>.<release>`".format(
```

Comment:
```suggestion
                "Must be in the form `<major>.<minor>` or `<major>.<minor>.<patch>`".format(
```

For consistency with the CLI docs above

Reply:
Yup - fixed.